tsdb/series_set.go

```diff
@@ -71,3 +72,28 @@ func (s *SeriesIDSet) Merge(others ...*SeriesIDSet) {
 	defer s.Unlock()
 	s.bitmap = roaring.FastOr(bms...)
 }
+
+// Diff deletes any bits set in other.
+func (s *SeriesIDSet) Diff(other *SeriesIDSet) {
+	other.RLock()
+	bm := other.bitmap
```

Comment:
I think you need an `RLock` for the duration of the operation, i.e.,

```go
func (s *SeriesIDSet) AndNot(other *SeriesIDSet) *SeriesIDSet {
	other.RLock()
	defer other.RUnlock()
	
        s.Lock()
	defer s.Unlock()
	s.bitmap =  roaring.AndNot(s.bitmap, other.bitmap)
}
```

Reply:
@e-dard I was wondering about that. The `SeriesIDSet.Merge()` didn't `RLock` while merging so I wasn't sure if it was a copy-on-write thing. I updated in 725117b.